### PR TITLE
fix: support docker create arguments from container.options (#1022)

### PR DESCRIPTION
**Pull request by alex-savchuk**
Important note: parsing of CLI options of docker is fetched from docker/cli sources explicitly - see docker_cli.go. 
Please confirm it's acceptable.

Fixes https://github.com/nektos/act/issues/711
Fixes https://github.com/nektos/act/issues/1022

### CODE REVIEW

File: pkg/container/docker_cli.go
Comment (alex-savchuk) [reply pkg/container/docker_cli.go:2]: Ok, thanks. Let's wait for others then.

Used golint version does not support "//nolint:all" so ignoring of this file is done by "//nolint:unparam,errcheck,depguard,deadcode,unused".
Also I exclude this file from megalinter checking too + fix all found lint issues.

**Comment by alex-savchuk** [comment]
It's a bit not clear for me what to do with codecov/project check.

I look at diff report at https://codecov.io/gh/nektos/act/compare/4f8da0a51cd44753f3e4ea0364877f6d2aa11f1d...9781bfbe18671a3529340150d12c3eb6c5e20c67/changes
 but don't understand how my small changes result into this decreasing.

Maybe we need to switch HEAD of coverage report to some modern version ?
Or I need to add tests for logger.go/run_context.go in PR?

File: pkg/container/docker_cli.go
Comment (KnisterPeter) [inline pkg/container/docker_cli.go:5]: Since this is a copy, we have to add the license to this file.
We are not allowed to copy their code as is without keeping the license.

**Review by KnisterPeter — CHANGES_REQUESTED** [review]
Besides the license, this looks fine to me. Great addition.

File: pkg/container/docker_cli.go
Comment (alex-savchuk) [reply pkg/container/docker_cli.go]: Add header to this file and DOCKER_LICENSE with full text.
Please check

**Comment by alex-savchuk** [comment]
I agree that it's better to copy tests too.
Let me do that

**Comment by alex-savchuk** [comment]
Added opts_test.go, required testdata and updated dependencies. Please take a look.
I hope it solves coverage problem with this PR. :)

**Comment by alex-savchuk** [comment]
After merge #711, #1022 are expected to be fixed.
Also passing of docker options may help to resolve #676, #569, #1091

**Comment by alex-savchuk** [comment]
@act-maintainers Now we have coverage, tests etc. Could you please take a look and approve/merge if everything is ok? :)

**Review by KnisterPeter — APPROVED** [review]
I really like it

**Review by KnisterPeter — APPROVED** [review]

**Review by cplee — APPROVED** [review]
Sorry for the delay - this was a large PR and took awhile to get through all the changes. Thanks for your contribution!

**Pull request merged**
